packages/@aws-cdk/aws-iam/lib/managed-policy.ts

```diff
@@ -43,6 +137,129 @@ export class ManagedPolicy {
     return new AwsManagedPolicy();
   }
 
-  protected constructor() {
+  /**
+   * Returns the ARN of this managed policy.
+   *
+   * @attribute
+   */
+  public readonly managedPolicyArn: string;
+
+  /**
+   * The policy document.
+   */
+  public readonly document = new PolicyDocument();
+
+  /**
+   * The name of this policy.
+   *
+   * @attribute
+   */
+  public readonly managedPolicyName: string;
+
+  /**
+   * The description of this policy.
+   *
+   * @attribute
+   */
+  public readonly description: string;
+
+  /**
+   * The path of this policy.
+   *
+   * @attribute
+   */
+  public readonly path: string;
+
+  private readonly roles = new Array<IRole>();
+  private readonly users = new Array<IUser>();
+  private readonly groups = new Array<IGroup>();
+
+  constructor(scope: Construct, id: string, props: ManagedPolicyProps = {}) {
+    super(scope, id, {
+      physicalName: props.managedPolicyName ||
+        // generatePolicyName will take the last 128 characters of the logical id since
+        // policy names are limited to 128. the last 8 chars are a stack-unique hash, so
+        // that shouod be sufficient to ensure uniqueness within a principal.
+        Lazy.stringValue({ produce: () => generatePolicyName(scope, resource.logicalId) }),
+    });
+
+    this.managedPolicyName = this.physicalName;
```

Comment:
You need to use `this.getResourceNameAttribute`

Reply:
This was also based off Policy, does that need to be changed too?